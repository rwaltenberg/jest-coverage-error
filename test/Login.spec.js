import { mount, createLocalVue } from '@vue/test-utils';
import VueBootstrap from 'bootstrap-vue';
import Login from '~/pages/login/index.vue';

const localVue = createLocalVue();
localVue.use(VueBootstrap);

const factory = (params = {}) => {
  return mount(Login, {
    ...params,
    localVue,
    mocks: {
      login: () => {},
      $mixpanel: () => {},
      $oauth: {
        logout: () => {},
      },
    },
  });
};

const loginTest = (name) => {
  test(`${name} login`, () => {
    const wrapper = factory();
    const loginMock = jest.fn();
    wrapper.vm.login = loginMock;

    const loginButton = wrapper.find(`.${name}`);

    loginButton.trigger('click').catch((error) => {
      throw error;
    });

    expect(loginMock.mock.calls.length).toBe(1);
    expect(loginMock).toHaveBeenCalledWith(name);
  });
};

describe('Login page', () => {
  test('mounts properly', () => {
    const wrapper = factory();
    expect(wrapper.vm).toBeTruthy();
  });
  test('renders properly', () => {
    const wrapper = factory();
    expect(wrapper.html()).toMatchSnapshot();
  });

  loginTest('linkedin');
  loginTest('salesforce');
  loginTest('google');
});
