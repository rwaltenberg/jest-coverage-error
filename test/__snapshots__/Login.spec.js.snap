// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login page renders properly 1`] = `
<div class="login-wrapper container">
  <div class="row">
    <div class="col login-action-item linkedin">
      Linkedin
    </div>
    <div class="col login-action-item salesforce">
      SFDC
    </div>
    <div class="col login-action-item google">
      Google
    </div>
  </div>
</div>
`;
